### ISSUE TITLE
EBERG QUBO Q40HD 4,1kW

### ISSUE THREAD

**Issue by SatarisGIT** [make-all/tuya-local#35]
Hello!

I have problem with my AC in tuya local integration.

When I adding it, I cant see "climate" option. I can only choose "simple_switch". If I do this - I can only turn on and off my AC.

There is any option which allow me to "manually" add missing features? (temperature control/mode set)

Thanks for any responses :)

**Comment by make-all** [comment]
There is no standard for tuya "climate" devices, so unless the device has already had a configuration file created for it, it will not work out of the box.

What is the brand and model of the device, and what messages do you see in the Home Assistant log?

**Comment by SatarisGIT** [comment]
Thanks for reply, its: EBERG QUBO Q40HD 4,1kW
I dont have any logs about this machine

**Comment by make-all** [comment]
When you add the device and it only offers "Simple Switch", there should be two WARNING level log messages output to the Home Assistant log (which you can access under Confguration).  Warnings are enabled by default, so unless you customized the logging, they should show up  there.  These logs contain the data that is output by your device, which is the first step in being able to recognize it.  The next step would be to figure out what each data point is for, and what range of values it can take, so a config can be made that makes full use of the device.

**Comment by make-all** [comment]
Unfortunately, without the info from the log file, there is not much that can be done. I cannot find any technical information about that model of AC that would help with adding support.

**Comment by SatarisGIT** [comment]
Thanks for response!

I've got this:

![image](https://user-images.githubusercontent.com/28608081/135266753-5c02d692-5cf3-4fbd-b5dc-dd4d775412f9.png)
```
Report this to https://github.com/make-all/tuya-local/issues/
14:12:59 – (OSTRZEŻENIA) Tuya Local (niestandardowa integracja)
Device matches simple_switch with quality of 10%. DPS: {'1': True, '2': 22, '3': 20, '4': 'hot', '5': 'middle', '19': 'c', '22': 0, '25': False, '30': False, '101': 'heat_s', 'updated_at': 1632917578.2792246}
14:12:59 – (OSTRZEŻENIA) Tuya Local (niestandardowa integracja)
```

Is there any possibility that I could add such an integration myself?

**Comment by make-all** [comment]
The device should be similar to other heatpump devices, so you could copy and modify one of the configs as a starting point.
The above log info is probably a good start to assigning dps to functionality, some more experimenting will be needed to discover all the possible values for all the fields - this may be easier if you get a basic config first.

dps 1 should be "power"
dps 2 and 3 will be temperature and current_temperature (I'm not sure which is which, but you should be able to tell quite easily by which matches the set temperature.
dps 4 is probably the hvac_mode, but will need mapping to the standard HA hvac_mode names (ie: hot -> heat). It will also need to be linked to the power dps, which the other heatpumps that support more than one mode will have examples of.
dps 5 may be fan_mode or swing_mode I would guess.  Again, they may need to be mapped to HA default values.  If you can't figure it out immediately, name it unknown_5 and just leave it for monitoring later.
dps 19 looks like a C/F selector - probably just report it for now.
dps 22 may be a timer, error code, or some other unknown value, for now I would just call it unknown_22 until it can be identified so the value can be monitored.
dps 25 and 30 look like switches of some sort, but again, you can name them unknown_25 and unknown_30 and just monitor until you figure them out.
dps 101 could be preset_mode, or maybe hvac_action, but if you can't identify which, just make it unknown_101 for monitorng.

**Comment by make-all** [comment]
OK.  I made the config based on info I found here: http://eberg.eu/klimatyzatory_przenosne/qubo-q40hd/ (including a link to a manual that was in English)
The webpage mentions "4 in 1 – chłodzenie, grzanie, osuszanie, wentylacja" which I took to mean it supports dehumidifying and fan (air purifying) modes as well as cold and hot, but the details above say just cold and hot.  The manual mentioned dehumidifying, but not the fan only mode, so they seem a bit confused over the features.
I guess the strange behaviour in dehumidifying mode is because it does not support that.  And the info in the manual about restriction of fan modes is probably wrong too then.
